spacy_llm/tasks/ner.py

```diff
@@ -168,26 +104,21 @@ def initialize(
             for initialization.
         nlp (Language): Language instance.
         labels (List[str]): Optional list of labels.
-        n_prompt_examples (int): How many prompt examples to infer from the Example objects.
-            0 by default. Takes all examples if set to -1.
         """
         if not labels:
             labels = list(self._label_dict.values())
-        infer_labels = not labels
-
-        if infer_labels:
-            labels = []
 
-        for eg in get_examples():
-            if infer_labels:
-                for ent in eg.reference.ents:
-                    labels.append(ent.label_)
+        examples = get_examples()
+        for eg in examples:
             if n_prompt_examples < 0 or len(self._prompt_examples) < n_prompt_examples:
                 self._prompt_examples.append(self._create_prompt_example(eg))
-
-        self._label_dict = {
-            self._normalizer(label): label for label in sorted(set(labels))
-        }
+        if not labels:
+            label_set = set()
+            for eg in examples:
```

Comment:
These `for` loops got rewritten, but they were using `infer_labels` on purpose so that we don't have to store all the `get_examples()` in a variable and we can actually just loop through them once instead of twice. I think the original implementation was more efficient.

Reply:
Fixed with refactoring.